beeline-core/src/main/java/io/honeycomb/beeline/tracing/SpanBuilderFactory.java

```diff
@@ -258,11 +258,10 @@ public Span build() {
             final String traceId = parentContext.isTraced() ? parentContext.getTraceId() : idProvider.generateId();
             final int sampleRate = traceSampler.sample(traceId);
             LOG.debug("Building span - sampling decision for traceId '{}' is '{}'", traceId, sampleRate);
-            return sampleRate > 0 ? createSendingSpan(sampleRate) : Span.getNoopInstance();
+            return sampleRate > 0 ? createSendingSpan(sampleRate, traceId) : Span.getNoopInstance();
         }
 
-        private Span createSendingSpan(final int sampleRate) {
-            final String traceId = parentContext.isTraced() ? parentContext.getTraceId() : idProvider.generateId();
+        private Span createSendingSpan(final int sampleRate, String traceId) {
             final PropagationContext context = new PropagationContext(
                 traceId,parentContext.getSpanId(), parentContext.getDataset(), parentContext.getTraceFields()
```

Comment:
can you fix this formatting here while you're at it? should be `traceId, parentcontext...` Thanks!

Reply:
Done